modules/internal/String.chpl

```diff
@@ -1099,14 +1107,19 @@ module String {
       }
 
       if trailing {
-        label outer for i in 0..#localThis.len by -1 {
-          for j in 0..#localChars.len {
-            if localThis.buff[i] == localChars.buff[j] {
-              end -= 1;
+        // Because we are working with codepoints whose starting byte index
+        // is not initially known, it is faster to work forward, assuming we
+        // are already past the end of the string, and then update the end
+        // point as we are proven wrong.
+        end = 0;
+        label outer for (thisChar, i, nbytes) in localThis[start..].ucharsIndexLen() {
```

Comment:
it seems to me ucharsIndexLen should be able to start at a particular byte offset (start).

Reply:
Yes, that will simplify this method.  Thanks.